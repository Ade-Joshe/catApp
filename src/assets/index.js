export * from "./allCatIcon";
export * from "./catsILikeIcon";
export * from "./emptyHeart";
export * from "./filledHeart";