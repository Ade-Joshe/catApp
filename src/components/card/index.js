export * from "./miniCard";
export * from "./emptyCard";
export * from "./largeCard";