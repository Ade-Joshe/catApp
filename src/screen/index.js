export * from "./favouriteScreen";
export * from "./homeScreen";