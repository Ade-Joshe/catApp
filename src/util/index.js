export * from "./pixelConverter";
export * from "./theme";
export * from "./store";