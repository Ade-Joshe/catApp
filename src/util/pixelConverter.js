import { PixelRatio, Dimensions } from 'react-native';

const BASE = {
    HEIGHT: 'height',
    WIDTH: 'width'
};

const { width, height } = Dimensions.get('window');
const baseScreen = {
    WIDTH: 375,
    HEIGHT: 812
};

const widthBaseScale = width / baseScreen.WIDTH;
const heightBaseScale = height / baseScreen.HEIGHT;

export const calcDevicePx = (px, base = BASE.WIDTH) => {
    const newSize = base === BASE.HEIGHT ? px * heightBaseScale : px * widthBaseScale;
    return Math.round(PixelRatio.roundToNearestPixel(newSize));
};

//for width  pixel
export const widthPixel = (size) => {
    return calcDevicePx(size, BASE.WIDTH);
};
//for height  pixel
export const heightPixel = (size) => {
    return calcDevicePx(size, BASE.HEIGHT);
};

//for font  pixel
export const fontPixel = (size) => {
    return widthPixel(size);
};
