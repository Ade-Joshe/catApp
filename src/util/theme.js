import { createContext } from "react";

export const theme = {
    colors: {
        black: "#212227",
        red: "#DE0202",
        grey: "#d3d3d4"
    }
};

export const ThemeContext = createContext(theme);