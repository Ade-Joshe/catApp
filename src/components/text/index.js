import React, { useContext } from 'react'
import { StyleSheet, Text } from 'react-native'
import { fontPixel, heightPixel, ThemeContext } from '../../util'

export const PageTitle = ({ children }) => {
    const { colors } = useContext(ThemeContext);
    return <Text style={[styles.title, { color: colors.black, }]}>{children}</Text>
};

const styles = StyleSheet.create({
    title: { fontSize: fontPixel(16), marginVertical: heightPixel(25), fontWeight: "600" }
})
