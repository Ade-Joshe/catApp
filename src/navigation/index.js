import React, { useContext } from "react"
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';
import { NavigationContainer } from "@react-navigation/native";
import { HomeScreen, FavouriteScreen } from '../screen';
import { AllCatIcon, CatsILikeIcon } from "../assets";
import { fontPixel, ThemeContext } from "../util";

const Tab = createBottomTabNavigator();

export function RootNavigation() {

    const { colors } = useContext(ThemeContext);

    return (
        <NavigationContainer>
            <Tab.Navigator
                screenOptions={({ route }) => ({
                    unmountOnBlur: true,
                    headerShown: false,
                    lazy: true,
                    tabBarActiveTintColor: colors.black,
                    tabBarInactiveTintColor: colors.grey,
                    tabBarStyle: { height: 67 },
                    tabBarLabelStyle: { fontSize: fontPixel(15) },
                    tabBarLabelPosition: 'below-icon',
                    tabBarIcon: ({ focused }) => {
                        if (route.name === 'All Cats') return <AllCatIcon color={focused ? colors.black : colors.grey} />;
                        else if (route.name === 'Cats I Like') return <CatsILikeIcon color={focused ? colors.black : colors.grey} />;
                        else return;
                    }
                })}
            >
                <Tab.Screen name="All Cats" component={HomeScreen} />
                <Tab.Screen name="Cats I Like" component={FavouriteScreen} />
            </Tab.Navigator>
        </NavigationContainer >
    );
}