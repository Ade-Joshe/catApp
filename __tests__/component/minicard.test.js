import React from "react"
import { render, fireEvent } from '@testing-library/react-native';
import { MiniCard } from "../../src/components";
import { create } from "react-test-renderer";
import { AppContext } from "../../src/util";

const mockData = {
    name: "Bengal",
    id: "dfsd",
    image: { url: "https://cdn2.thecatapi.com/images/mj.jpg" }
};

// renders
it("renders correctly", () => {

    const { getAllByText } = render(
        <AppContext.Provider value={{ faveCats: [], setFaveCats: () => { } }}>
            <MiniCard data={mockData} />
        </AppContext.Provider>
    );

    expect(getAllByText("Bengal")).toBeTruthy();
});

// toggleFavourite
it("renders correctly", async () => {

    let mock = [mockData];

    const { getByTestId, toJSON } = render(
        <AppContext.Provider value={{ faveCats: mock, setFaveCats: () => { } }}>
            <MiniCard data={mockData} />
        </AppContext.Provider>
    );

    await fireEvent.press(getByTestId("favourite-btn"));
    await expect(toJSON()).toMatchSnapshot();
});

it("snapshot testing for default loading", () => {
    let component = create(
        <AppContext.Provider value={{ faveCats: [], setFaveCats: () => { } }}>
            <MiniCard data={mockData} />
        </AppContext.Provider>);

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});

