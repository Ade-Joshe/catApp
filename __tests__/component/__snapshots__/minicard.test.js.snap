// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 17,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://cdn2.thecatapi.com/images/mj.jpg",
      }
    }
    style={
      Object {
        "backgroundColor": "#d3d3d4",
        "borderRadius": 20,
        "height": 80,
        "width": 80,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#212227",
          },
          Object {
            "fontSize": 32,
            "fontWeight": "400",
            "marginLeft": 30,
            "paddingVertical": 25,
          },
        ]
      }
    >
      Bengal
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 60,
          "opacity": 1,
          "padding": 20,
        }
      }
      testID="favourite-btn"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="18"
        bbWidth="18"
        fill="none"
        focusable={false}
        height="18"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 18,
              "width": 18,
            },
          ]
        }
        vbHeight={18}
        vbWidth={18}
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M15.63 3.458a4.124 4.124 0 00-5.835 0L9 4.253l-.795-.795A4.126 4.126 0 002.37 9.293l.795.795L9 15.922l5.835-5.835.795-.795a4.124 4.124 0 000-5.835z"
            fill={4292739586}
            propList={
              Array [
                "fill",
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={4292739586}
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="1.5"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;

exports[`snapshot testing for default loading 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "marginVertical": 17,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://cdn2.thecatapi.com/images/mj.jpg",
      }
    }
    style={
      Object {
        "backgroundColor": "#d3d3d4",
        "borderRadius": 20,
        "height": 80,
        "width": 80,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#212227",
          },
          Object {
            "fontSize": 32,
            "fontWeight": "400",
            "marginLeft": 30,
            "paddingVertical": 25,
          },
        ]
      }
    >
      Bengal
    </Text>
    <View
      accessible={true}
      collapsable={false}
      focusable={true}
      nativeID="animatedComponent"
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderRadius": 60,
          "opacity": 1,
          "padding": 20,
        }
      }
      testID="favourite-btn"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="18"
        bbWidth="18"
        fill="none"
        focusable={false}
        height="18"
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            Object {
              "flex": 0,
              "height": 18,
              "width": 18,
            },
          ]
        }
        vbHeight={18}
        vbWidth={18}
        width="18"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={null}
          propList={
            Array [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M15.63 3.458a4.125 4.125 0 00-5.835 0L9 4.253l-.795-.795A4.126 4.126 0 002.37 9.293l.795.795L9 15.922l5.835-5.835.795-.795a4.125 4.125 0 000-5.835v0z"
            opacity={0.3}
            propList={
              Array [
                "stroke",
                "strokeWidth",
                "strokeLinecap",
                "strokeLinejoin",
              ]
            }
            stroke={4280361511}
            strokeLinecap={1}
            strokeLinejoin={1}
            strokeWidth="1.5"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;
